src/plugins/intel_cpu/src/nodes/random_uniform.cpp

```diff
@@ -399,135 +528,220 @@ std::pair<uint64_t, uint64_t> RandomUniform::computePhilox(void* out, size_t out
     return { n_state, counter_state };
 }
 
-////////////// STL algo ///////////////
-void RandomUniform::computeStl(void* out, size_t work_amount) {
-    switch (m_output_prc) {
-        case element::f32: {
-            generateData<float, std::uniform_real_distribution<float>>(
-                    std::uniform_real_distribution<float>{m_min_val.f32, m_max_val.f32}, out, work_amount);
-        } break;
-        case element::i32: {
-            generateData<int32_t, std::uniform_int_distribution<int32_t>>(
-                    std::uniform_int_distribution<int32_t>{m_min_val.i32, m_max_val.i32}, out, work_amount);
-        } break;
-        case element::i64: {
-            generateData<int64_t, std::uniform_int_distribution<int64_t>>(
-                    std::uniform_int_distribution<int64_t>{m_min_val.i64, m_max_val.i64}, out, work_amount);
-        } break;
-        default:
-            THROW_CPU_NODE_ERR("has unsupported output type: ", m_output_prc);
+////////////// MERSENNE algo ///////////////
+
+namespace {
+
+constexpr int32_t MERSENNE_STATE_N = 624;
```

Comment:
Is there any article which explains these magic numbers? Can we leave the comment here with the link  to the article as it's done for the philox algorithm?

Reply:
Added a link! The values are from the algorithm itself, as it has a precisely chosen size of state variables, look-forward offsets, etc. https://dl.acm.org/doi/10.1145/272991.272995